python/lsst/ts/cRIOpy/VMS/CSCPSDWidget.py

```diff
@@ -0,0 +1,178 @@
+# This file is part of cRIO/VMS GUI.
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org). See the COPYRIGHT file at the top - level directory
+# of this distribution for details of code ownership.
+#
+# This program is free software : you can redistribute it and / or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["CSCPSDWidget"]
+
+import math
+
+import numpy as np
+from PySide2.QtCore import Qt, Slot, QPointF
+from PySide2.QtCharts import QtCharts
+
+from ..GUI import AbstractChart
+from .ChartView import ChartView
+from ..GUI.CustomLabels import DockWindow
+from .Unit import units, coefficients
+
+
+class CSCPSDWidget(DockWindow):
+    """Display CSC calculated PSD.
+
+    Parameters
+    ----------
+    title : `str`
+        QDockWidget title and object name.
+    toolBar : `ToolBar`
+        Provides getFrequencyRange() method.
+    psd : `QSignal`
+        PSD SAL signal.
+    num_sensor : `int`
+        Number of sensors.
+    channles : `[int]`, optional
+        Channels to plot. Defaults to empty array, user should select channels
+        from context menu.
+    """
+
+    def __init__(self, title, toolBar, psd, num_sensor: int, channels: [int] = []):
+        super().__init__(title)
+        self.toolBar = toolBar
+
+        self.chart = AbstractChart()
+
+        self.chartView = ChartView(self.chart, QtCharts.QLineSeries)
+        self.chartView.updateMaxSensor(num_sensor)
+        self.chartView.axisChanged.connect(self.axisChanged)
+        self.chartView.unitChanged.connect(self.unitChanged)
+        for channel in channels:
+            self.chartView.addSerie(str(channel[0]) + " " + channel[1])
+
+        self.coefficient = 1
```

Comment:
What is this coefficient?

Reply:
Unit conversion factor. In fact it wasn't used. I fixed the code to use it properly and add a docstring.